include/swift/AST/DiagnosticsSIL.def

```diff
@@ -371,6 +371,23 @@ ERROR(possible_return_before_yield, none,
 
 NOTE(one_yield, none, "yield along one path is here", ())
 
+NOTE(false_branch_doesnt_yield, none,
+    "must yield when the condition is false", ())
```

Comment:
For the notes, I think you should be informative rather than prescriptive.

How about "missing yield when the condition is false", etc.?